### PR TITLE
Keras support update for TensorFlow 2.0 compatibility

### PR DESCRIPTION
**Pull request by juntai-zheng**
## What changes are proposed in this pull request?
 
MLflow Keras model saving, loading, and logging has been updated to be compatible with TensorFlow 2.0. This mainly included work on creating alternate code paths that removed Graphs and Sessions from the logic. Other fixes include:

- Conda-forge has issues installing the latest Keras and TensorFlow modules (namely, conda-forge does not have them). When the newest versions are required when creating a conda.yaml environment file, the dependencies are moved to pip dependencies instead. 
- TensorFlow becomes extremely buggy when `tensorflow_core` is imported before `tensorflow` (as previously in `_load_pyfunc()`). This has been changed to import TensorFlow first.
-  Fixing flakiness of `test_keras_model_export.py` tests by changing `np.assert_array_almost_equal()`, which compared numbers to decimal precision, to `np.allclose()`, which compares numbers relatively. Flakiness occurred because the numbers being compared are huge, around `1e25`.
- Fixing flakiness of `test_keras_model_export.py` tests by including `encoding="utf-8"` as an argument for `pd.read_json()`, since the function would occasionally error out with `ValueError: Unexpected character found when decoding object value`. These fixes should address #1095.
- Changing the deprecated `mlflow.pyfunc.load_pyfunc()` to `mlflow.pyfunc.load_model()`.
- Adding back Keras tests back to Travis.
 
## How is this patch tested?
 
Via `test_keras_model_export.py` and manual testing with different Keras/TF versions. Also used pytest-repeat in order to test for test flakiness.
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
MLflow now supports usage of Keras models with TensorFlow 2.0.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [X] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/keras.py
Comment (smurching) [inline mlflow/keras.py]:
```suggestion
    pip_deps = []
```

File: mlflow/keras.py
Comment (smurching) [inline mlflow/keras.py]:
```suggestion
            keras_dependency.append("keras=={}".format(keras_module.__version__))
```

File: mlflow/keras.py
Comment (smurching) [inline mlflow/keras.py]:
```suggestion
            pip_deps.append("keras=={}".format(keras_module.__version__))
```

File: mlflow/keras.py
Comment (smurching) [inline mlflow/keras.py]:
```suggestion
        pip_deps.append("cloudpickle=={}".format(cloudpickle.__version__))
```

File: travis/run-large-python-tests.sh
Comment (smurching) [inline travis/run-large-python-tests.sh]: Hm I guess this gives us a bit more coverage (to install old Keras first & run tests, then install the new Keras) - but I think it's fine to just test against the new Keras (e.g. just update https://github.com/mlflow/mlflow/blob/master/travis/large-requirements.txt#L27 to Keras 2.3.1). We should in a separate follow-up find a long-term fix for correctly capturing which dependencies the user installed via conda vs pip, and recording that info in the conda.yaml saved with the model (would you mind filing a GitHub issue to track that work)?

**Review by smurching — COMMENTED** [review]
Mostly looks good, just some small comments!

File: travis/run-large-python-tests.sh
Comment (juntai-zheng) [reply travis/run-large-python-tests.sh]: I can make the GitHub issue! In regards to changing the default Keras testing version, a keras_autolog test fails due to one of the metric's name being changed from `'acc'` to `'accuracy'` in Keras 2.3.1. I can go ahead and make the quick version check as part of this PR so we don't have to change the `run-large-python-tests.sh` twice unnecessarily.

File: tests/keras_autolog/test_keras_autolog.py
Comment (smurching) [inline tests/keras_autolog/test_keras_autolog.py]: Maybe we don't need this check anymore now that we're only testing Keras 2.3.1? Can simplify the code & probably fix the test that way :)

**Review by smurching — APPROVED** [review]
LGTM! Just need to fix one test, after we do that let's merge

**Pull request merged**
